### PR TITLE
O3-1400: Create unit tests for the subscription component

### PR DESCRIPTION
**Pull request by Piumal1999**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

Added the test configuration and wrote unit tests for the root component and subscription component

## Screenshots

image

## Issue

https://issues.openmrs.org/browse/O3-1400

## Other

*None.*

### CODE REVIEW

File: packages/esm-admin-openconceptlab-app/src/subscription/subscription.resource.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { openmrsFetch } from '@openmrs/esm-framework';
 import useSWR from 'swr';
 import { Subscription } from '../types';
-import { isNil } from 'lodash-es';
+import { isNil } from 'lodash';
```
Comment (ibacher) [inline packages/esm-admin-openconceptlab-app/src/subscription/subscription.resource.ts:4]:
```suggestion
import isNil from 'lodash-es/isNil';
```

File: __mocks__/react-i18next.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  useTranslation: () => useMock,
+
+  // mock if needed
+  I18nextProvider: reactI18next.I18nextProvider,
+  initReactI18next: reactI18next.initReactI18next,
+  setDefaults: reactI18next.setDefaults,
+  getDefaults: reactI18next.getDefaults,
+  setI18n: reactI18next.setI18n,
+  getI18n: reactI18next.getI18n,
+};
```
Comment (ibacher) [inline __mocks__/react-i18next.js:49]: It's always a good idea to have a blank line at the end of a file (this makes the algorithm the `diff` tool works a little more reliable).

```suggestion
};

```

File: __mocks__/react-i18next.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if (typeof child === 'object' && !isElement) {
+      return Object.keys(child).reduce((str, childKey) => `${str}${child[childKey]}`, '');
+    }
+
+    return child;
+  });
+};
+
+const useMock = [(k) => k, {}];
+useMock.t = (k, o) => (o && o.defaultValue) || (typeof o === 'string' ? o : k);
```
[the remark this answers is not in the document]
Comment (ibacher) [reply __mocks__/react-i18next.js:33]: I think it's actually fine to check that the key is displayed in the unit tests. To actually test the translations, we can use e2e tests.

**Comment by ibacher** [comment]
@Piumal1999 Is this ready to be reviewed? If so, can you mark it as "Ready for review"?

**Comment by Piumal1999** [comment]
> @Piumal1999 Is this ready to be reviewed? If so, can you mark it as "Ready for review"?

Yeah @ibacher, I added a few more tests. Now this is ready for review.

**Review by ibacher — APPROVED** [review]
This looks good. A couple of suggestions for improvement, but those can be integrated into another PR.

**Comment by Piumal1999** [comment]
Thank you for the suggesstions @ibacher. I included those changes in PR #5.

**Pull request merged**